Sources/LinuxPlatform/Linux.swift

```diff
@@ -481,98 +535,62 @@ public struct Linux: Platform {
             if disableConfirmation {
                 throw Error(message: message)
             } else {
                 print(message)
             }
-            return self.manualSelectPlatform(platformPretty)
+            return await self.manualSelectPlatform(platformPretty)
         }
 
-        if (id + idlike).contains("amzn") {
-            guard let versionID = versionID, versionID == "2" else {
+        if (id + (idlike ?? "")).contains("amzn") {
+            guard versionID == "2" else {
                 let message = "Unsupported version of Amazon Linux"
                 if disableConfirmation {
                     throw Error(message: message)
                 } else {
                     print(message)
                 }
-                return self.manualSelectPlatform(platformPretty)
+                return await self.manualSelectPlatform(platformPretty)
             }
 
-            return PlatformDefinition(name: "amazonlinux2", nameFull: "amazonlinux2", namePretty: "Amazon Linux 2")
-        } else if (id + idlike).contains("ubuntu") {
-            if ubuntuCodeName == "jammy" {
-                return PlatformDefinition(name: "ubuntu2204", nameFull: "ubuntu22.04", namePretty: "Ubuntu 22.04")
-            } else if ubuntuCodeName == "focal" {
-                return PlatformDefinition(name: "ubuntu2004", nameFull: "ubuntu20.04", namePretty: "Ubuntu 20.04")
-            } else if ubuntuCodeName == "bionic" {
-                return PlatformDefinition(name: "ubuntu1804", nameFull: "ubuntu18.04", namePretty: "Ubuntu 18.04")
-            } else {
-                let message = "Unsupported version of Ubuntu Linux"
-                if disableConfirmation {
-                    throw Error(message: message)
-                } else {
-                    print(message)
-                }
-                return self.manualSelectPlatform(platformPretty)
-            }
-        } else if (id + idlike).contains("rhel") {
-            guard let versionID = versionID, versionID.hasPrefix("9") else {
+            return PlatformDefinition.amazonlinux2
+        } else if (id + (idlike ?? "")).contains("rhel") {
+            guard versionID.hasPrefix("9") else {
                 let message = "Unsupported version of RHEL"
                 if disableConfirmation {
                     throw Error(message: message)
                 } else {
                     print(message)
                 }
-                return self.manualSelectPlatform(platformPretty)
+                return await self.manualSelectPlatform(platformPretty)
             }
 
-            return PlatformDefinition(name: "ubi9", nameFull: "ubi9", namePretty: "RHEL 9")
+            return PlatformDefinition.rhel9
+        } else if let pd = [PlatformDefinition.ubuntu1804, .ubuntu2004, .ubuntu2204, .ubuntu2310, .ubuntu2404, .debian12, .fedora39].first(where: { $0.name == id + versionID }) {
```

Comment:
If we're gonna maintain a static data structure here, might as well make it a dictionary so we get better performance on look ups? Any thoughts? It's not really a big deal but since we're going to maintain the list of linux versions here might be worth it